docs/faq.md

```diff
@@ -73,11 +73,26 @@ Embed a special version of Semgrep Playground with an `iframe`. The source is `h
 <iframe title="Semgrep example no prints" src="https://semgrep.dev/embed/editor?snippet=KPzL" width="100%" height="432" frameborder="0"></iframe>
 ```
 
+<!-- vale off -->
+### How does Semgrep go "beyond regex"?
+<!-- vale on -->
+
+Semgrep is semantic grep for code: it understands the **structure of code** and builds a syntax tree to search for matches. Where `grep "2"` only matches the exact string `2`, Semgrep matches other equivalent forms, such as [`x = 1; y = x + 1`](https://semgrep.dev/playground/s/5rKgj) when searching for `2`. Semgrep's [pattern syntax](/writing-rules/pattern-syntax/) provides specific mechanisms to fine-tune matches, such as the [ellipsis operator](/writing-rules/pattern-syntax#ellipsis-operator) and [metavariables](/writing-rules/pattern-syntax#metavariables).
+
+See the following rule for a more complex example illustrating Semgrep features:
+
+<iframe title="Go dangerous method usage" src="https://semgrep.dev/embed/editor?snippet=go.gorm.security.audit.gorm-dangerous-methods-usage.gorm-dangerous-method-usage" width="100%" height="432" frameborder="0"></iframe>
+
+- It uses [typed metavariables](writing-rules/pattern-syntax/#typed-metavariables) so it can specify the type `http.Request`.
+- In the sink, the rule tracks imports down to function usage.
+- In the sanitizer, it removes type aware Booleans and a string convert function.
```

Comment:
```suggestion
- In the sanitizer, it removes type-aware Booleans and a string convert function.
```

Reply:
I believe "type-aware" (with the hyphen) is the preferred form, also because it prevents readers from misreading aware as modifying Booleans on its own, i.e. we hyphenate to make it clear that "type" and "aware" go together.